polkadot/node/subsystem-util/src/lib.rs

```diff
@@ -440,3 +439,34 @@ impl Validator {
 		Signed::sign(&keystore, payload, &self.signing_context, self.index, &self.key)
 	}
 }
+
+/// Return a mapping of the availability chunk indices(`ValidatorIndex -> ChunkIndex`).
+///
+/// The vector indices represent validator indices.
+/// If it's enabled, shuffle the indices using the `BlockNumber` as a randomness seed, so that other
+/// validators have a common view of the shuffle at a given block height.
+/// Otherwise, return the identity vector.
+pub fn availability_chunk_indices(
+	maybe_client_features: Option<ClientFeatures>,
+	block_number: BlockNumber,
+	n_validators: usize,
+) -> Vec<ChunkIndex> {
+	let mut indices: Vec<_> = (0..n_validators)
+		.map(|i| ValidatorIndex(u32::try_from(i).expect("validator numbers should not exceed u32")))
+		.collect();
+
+	if let Some(features) = maybe_client_features {
+		if features.contains(ClientFeatures::AVAILABILITY_CHUNK_SHUFFLING) {
+			let seed = block_number.to_be_bytes();
```

Comment:
It's probably fine either way.  The trade off is this:

Imagine an adversary who wants everyone to do full reconstructions.  It's clear they must control the backers.  If we've a fixed shuffle independent of core, then they can do this when both (a) they control the backers, and (b) they have 4-5 nodes spread around fairly evenly in the current global shuffle.  If our shuffle depends upon the core, then the adversary should be slightly bigger to have good odds of landing in the first third.  Does this adversary matter?  I donno..

We might distribute load slightly more evenly by placing the systematic chunks equidistant, which requires our shuffle be independent of core.  It's possible this gets washed out by natural variance among parachain work loads.  Again I donno..

I think the actual fetching code matters way more.

Reply:
If we use on-chain randomness in the shuffle, then we'd use the on-chain randomness from roughly when we elect the validators, no?  If so, how can they have different views, without having different views abut the election?

Do you mean because some message might not mention the core?

Reply:
I was thinking of an edge-case scenario when there are two blocks at the same height, but in different sessions. This can happen e.g. due to dispute reversions. We would cache the mapping per block height, but the validator set could in theory be different. But this will not really cause issues.

Reply:
If we've a fork, then the validators won't have the chunks anyways, since they know they're not working on that fork.  Or they'll work on both forks?

Availability votes cannot be moved between chains, right?

Reply:
Same availability votes can be included in both forks. In fact, this is what happened with disputes issue on Kusama and https://github.com/paritytech/polkadot-sdk/pull/1358 had to account for.

Reply:
We've other places where we require that forks cannot outlive a sessions, so whatever.  If however we want to tolerate long forks, then we'd need to change that, by adding a block reference into the vote or whatever.